libpod/container_internal_linux.go

```diff
@@ -1786,6 +1795,31 @@ func (c *Container) getHosts() string {
 			}
 		}
 	}
+
+	// Add gateway entry
+	var depCtr *Container
+	if c.config.NetNsCtr != "" {
+		// ignoring the error because there isn't anything to do
+		depCtr, _ = c.getRootNetNsDepCtr()
+	} else if len(c.state.NetworkStatus) != 0 {
+		depCtr = c
+	} else {
```

Comment:
This last block is not necessary.  depCtr will be nil by default.

Reply:
I'm not sure syntactically how to get rid of that block :grimacing: 

I added it because I wanted to ensure mutual exclusion between shared container network namespace configurations and containers with network namespaces created for them. Basically I didn't want the situation where

```
        var depCtr *Container
	if c.config.NetNsCtr != "" { // **TRUE**
		// ignoring the error because there isn't anything to do
		depCtr, _ = c.getRootNetNsDepCtr()
	} 
	if len(c.state.NetworkStatus) != 0 { // **TRUE**
		depCtr = c
	}
```

But If that is not possible / something to worry about I can make them both `if`'s and remove the unnecessary block

Reply:
@rhatdan just in case you didn't see my response.

Reply:
Yes, that is fine.